README.md

```diff
@@ -977,48 +977,20 @@ Here is a demo of an interactive session running on Pixel 5 phone:
 
 https://user-images.githubusercontent.com/271616/225014776-1d567049-ad71-4ef2-b050-55b0b3b9274c.mp4
 
-#### Building the Project using Termux (F-Droid)
-Termux from F-Droid offers an alternative route to execute the project on an Android device. This method empowers you to construct the project right from within the terminal, negating the requirement for a rooted device or SD Card.
-
-Outlined below are the directives for installing the project using OpenBLAS and CLBlast. This combination is specifically designed to deliver peak performance on recent devices that feature a GPU.
-
-If you opt to utilize OpenBLAS, you'll need to install the corresponding package.
+#### Build on Android using Termux (F-Droid)
+F-Droid Termux is an alternative to execute `llama.cpp` on an Android device(*no root required*).
 ```
-apt install libopenblas
+apt update && apt upgrade -y
+apt install git
 ```
 
-Subsequently, if you decide to incorporate CLBlast, you'll first need to install the requisite OpenCL packages:
+Due to permission limitations in the Android API, it's essential to move your model inside the `~/` directory for best performance:
```

Comment:
This sentence is a bit to vague for me. What limitations exactly?

Reply:
Let's change it. Root folders aren't accessible on Android, ie. Android/data for system file security.

Termux has a problem specifically with storage/downloads path. ie., in Termux: `chmod +x` to a file in the downloads path **doesn't work** due to the limits I'm referring to in Android OS.

Termux `~/` paths to `/data/data/com.termux/files/home`, which **is accessible**. It supports file access modes, executable permission and special files like symbolic links. [source](https://github.com/termux/termux-packages/wiki/Termux-file-system-layout#home-directory).

Reply:
Ok, I understand that there are permission issues that prevent running and accessing certain folders from termux, but from what I understand the downloads directory should be accessible, but this says that it affects performance. Why is that?

Reply:
> from what I understand the downloads directory should be accessible, bu this says that it affects performance

I don't know exactly why the sandbox Termux creates doesn't perform as well when loading a model from inside the downloads folder compared to Termux root `~/`. I'm open to finding out.

It's clear `./main -m ~/model.gguf` is **absolutely faster** than `./main -m /data/data/com.termux/files/home/storage/downloads/model.gguf`.

I haven't seen any issue where an Android user loaded from downloads, and it outpeforms loading from `~/`.

@l3utterfly Have any insight as to why models are lower performance when loaded from downloads?

Reply:
@slaren The performance degrade is due to Android Storage Access Framework API. 

Termux, or [Layla](https://play.google.com/store/apps/details?id=com.laylalite) will navigate to shared storage(Android, DCIM, Movies, Pictures, Downloads, ect..), but there's overhead compared to running `llama.cpp` from within the apps native space. Layla also solves this by automatically moving the model to it's own internal directory:
![Screenshot_20240503_201017](https://github.com/ggerganov/llama.cpp/assets/21167443/e7963262-9459-481b-b90a-93d262ecd7cf)

Anyway, I included the detail so users don't blame `llama.cpp` for bad performance after loading from the wrong spot. Maybe it's better to give the direction without a reason, like with https://github.com/ggerganov/llama.cpp/pull/7016/commits/624a689b68d177e194455f6635228f38b7e3079f

Let me know what you think!

Reply:
Thanks, I looked up SAF and I think I understand the issue, filesystem access to shared directories is emulated through a java module that seems to be extremely slow. This should only affect the load time of the model though, but performance during inference should be the same.